operations/generate-ri-report.py

```diff
@@ -0,0 +1,79 @@
+import glob
+import json
+import re
+from argparse import Namespace, ArgumentParser
+from logging import Formatter
+from os import path
+from typing import Dict
+import importlib
+import subprocess
+import sys
+
+LOG_FORMAT = Formatter(fmt="[%(asctime)s %(levelname)s %(module)s.%(name)s.%(funcName)s:%(lineno)d] %(message)s")
+
+EXTERNAL_DEPENDENCIES = {
+    "logzero": "logzero",
+    "boto3": "boto3"
+}
+
+
+class Bootstrap(object):
+    """
+    Bootstrap a new set of packages not included in the Python3 stdlib.
+    """
+
+    def __init__(self, packages: dict, **kwargs):
+        self.__dict__.update(**kwargs)
+        self.packages = packages
+        self._init()
+
+    def _init(self):
+        for package, import_name in self.packages.items():
+            subprocess.call([sys.executable, "-m", "pip", "install", package])
+            globals()[package] = importlib.import_module(import_name)
+
+
+def args() -> Namespace:
+    args_root = ArgumentParser(usage="Generate a report of deployed VM types for RI purchasing.")
+    args_root.add_argument("--debug", help="Enable debug logging.", action="store_true")
+    args_root.add_argument("--aws-access-key-id", help="AWS access key for the account you want to report on.",
+                           type=str)
+    args_root.add_argument("--aws-secret-access-key", help="Matching AWS secret account key.",
+                           type=str)
+
+    for g in args_root._action_groups:
+        g._group_actions.sort(key=lambda x: x.dest)
+
+    return args_root.parse_args()
+
+
+def main():
+    Bootstrap(packages=EXTERNAL_DEPENDENCIES)
+    cmd_args = args()
+
+    from logzero import setup_logger
```

Comment:
the python logger is more than sufficient for a 100-line script, especially since boto3 uses it as well.